test/views/pages/map/dynamic_map_test.dart

```diff
@@ -0,0 +1,286 @@
+    /// have selected during the test, [expectedPinDelta] is the expected difference in the number
+    /// of pins after the protocol is run, and [protocol] is the function that runs the protocol
+    /// (this is where one would delete a post, or create a new post, for instance).
+    void testNavigation({
+      required String testTitle,
+      required MapSelectionOptions optionToTest,
+      required int expectedPinDelta,
+      required Future<void> Function(
+        WidgetTester tester,
+        ProviderContainer container,
+      ) protocol,
+    }) {
+      testWidgets(testTitle, (tester) async {
+        final container = await beginTest(tester);
+
+        // Tap on the option chip we want for the test
+        final chip = find.byKey(
+          MapSelectionOptionChips.optionChipKeys[optionToTest]!,
+        );
+        expect(chip, findsOneWidget);
+        await tester.tap(chip);
+
+        // Verify we have the correct number of pins
+        final expectedPins =
+            expectedPostsForOption[optionToTest] ?? List.empty();
+        final pinsStart = await container.read(mapPinViewModelProvider.future);
+        expect(pinsStart, hasLength(expectedPins.length));
+
+        // Run the protocol
+        await protocol(tester, container);
+
+        // Verify the final state, the option should not have changed
+        final currentOption = container.read(
+          mapSelectionOptionsViewModelProvider,
+        );
+        expect(currentOption, equals(optionToTest));
+
+        // The pins should have the correct number
+        final pinsEnd = await container.read(mapPinViewModelProvider.future);
+        expect(pinsEnd, hasLength(expectedPins.length + expectedPinDelta));
+      });
+    }
+
+    testNavigation(
+      testTitle: "Pins refresh after post creation",
+      optionToTest: MapSelectionOptions.nearby,
+      expectedPinDelta: 1,
+      protocol: (tester, container) async {
+        await tester.tap(find.text("New post"));
+        final addPostButton = find.text("New post");
+        expect(addPostButton, findsOneWidget);
+        await tester.tap(addPostButton);
+        await tester.pumpAndSettle();
+
+        final titleForm = find.byKey(NewPostForm.titleFieldKey);
+        expect(titleForm, findsOneWidget);
+        await tester.enterText(titleForm, "Title");
+
+        final bodyForm = find.byKey(NewPostForm.bodyFieldKey);
+        expect(bodyForm, findsOneWidget);
+        await tester.enterText(bodyForm, "Body");
+
+        final submitButton = find.byKey(NewPostForm.postButtonKey);
+        expect(submitButton, findsOneWidget);
+        await tester.tap(submitButton);
+
+        await tester.pumpAndSettle();
+      },
+    );
+
+    testNavigation(
+      testTitle: "Pins refresh after post deletion",
+      optionToTest: MapSelectionOptions.myPosts,
+      expectedPinDelta: -1,
+      protocol: (tester, container) async {
+        final profileButton = find.byType(UserAvatar);
```

Comment:
I think that we could use the key that is present on the user avatar.

```suggestion
        final profileButton = find.byKey(HomeTopBar.profilePictureKey);
```

Reply:
Fixed in 5f038357a46beb16c994a21b3577faaf0e59d7a0.